lsp/server/src/server.ts

```diff
@@ -229,9 +329,8 @@ async function dockerContainer(folder: WorkspaceFolder): Promise<void> {
 				}
 			}
 		};
-		const sa4uConfig = await Promise.all([getSA4UConfig(), getIgnoredFiles()]).then((value) => {return {compilationDir: value[0].compilationDir, ignore: value[1].ignoreFilesCommand};});
-		console.log(`using compile dir: ${sa4uConfig.compilationDir}`);
-		const child = exec(`docker container run --rm --mount type=bind,source="${path}",target="/src/" --name sa4u_z3_server_${path.replace(/([^A-Za-z0-9]+)/g, '')} ${dockerImageName} -d True -c "${sa4uConfig.compilationDir}" ${sa4uConfig.ignore} -p /src/ex_prior.json -m /src/${messDef}`);
+		const sa4uConfig = await getSA4UConfig();
+		const child = exec(`docker container run ${(sa4uConfig.debugMaintainOnExit)?'':'--rm '}--mount type=bind,source="${path}",target="/src/" --name sa4u_z3_server_${path.replace(/([^A-Za-z0-9]+)/g, '')} ${sa4uConfig.returnParameters()}`);
```

Comment:
Why can't the logic to generate the `--rm` flag also reside in the `returnParameters` function?

Reply:
It can, it was a little inconvenient before since it would require ${path} to exist in the returned string as well.